### DISCUSSION TITLE
simple exist search

### DISCUSSION THREAD

**Discussion by RamilNurtdinov** [ClickHouse/ClickHouse#52532] [Q&A] [answered]
Hi

I need to perform quick search - if there is/are any record(s) in a particular table with requested text in primary indexed field.
I do not need the data, they are huge, I do not need data counts, this takes time.

I am doing this now with:
select * from table_of interest where job_md5 = 'XXXXX' limit 0,1

It would be nice to search PRIMARY KEY for the text of interest and report existence without entering the data like:
select exist from table_of interest where job_md5 = 'XXXXX'

Ramil
.

**Comment by den-crane** [comment, marked as answer]
You can use `select 1 from table_of interest where job_md5 = 'XXXXX' limit 1`
Clickhouse primary index is sparsed. It contains only some values (~1/8192) of table rows. 
It is impossible to get exact result using only primary index, need to read columns as well.
